lib/stackup/change_set.rb

```diff
@@ -73,8 +74,12 @@ def create(options = {})
           when /COMPLETE/
             return current.status
           when "FAILED"
-            logger.error(current.status_reason)
-            raise StackUpdateError, "change-set creation failed" if status == "FAILED"
+            if fail_change_set
+              return current.status_reason if current.execution_status == "UNAVAILABLE"
```

Comment:
This seems like it would not return at all if the stack got into a state where current.status was `FAILED` and current.execution_status was something other than `UNAVAILABLE` (where it should probably raise an exception). Also, it does no checking of whether the failure was actually caused by an empty change set.

Unfortunately there's no documented way to tell whether creating a change set failed because there were no changes or whether it failed for some other reason; probably the best solution would be to compare the human-readable error message, similar to what we do for the non-change-set case, https://github.com/realestate-com-au/stackup/blob/33e910f6451ff9f110f6e69c1cf4d5075545dc7c/lib/stackup/error_handling.rb#L22

Reply:
thanks @amcinnes 
I have changed the method of performing the check as suggested